crates/napi/src/threadsafe_function.rs

```diff
@@ -97,23 +98,83 @@ type_level_enum! {
 }
 
 struct ThreadsafeFunctionHandle {
-  raw: sys::napi_threadsafe_function,
+  raw: AtomicPtr<sys::napi_threadsafe_function__>,
   aborted: RwLock<bool>,
+  valid: AtomicBool,
   referred: AtomicBool,
 }
 
 unsafe impl Send for ThreadsafeFunctionHandle {}
 unsafe impl Sync for ThreadsafeFunctionHandle {}
 
+impl ThreadsafeFunctionHandle {
+  // called when the env is being destroyed
+  // then we can't call napi_****_threadsafe_function any more
+  unsafe extern "C" fn clearup(self_ptr: *mut c_void) {
```

Comment:
move it out of the `impl` block

Reply:
Is it because of mangling or something?

Reply:
This is mainly for consistency with the rest of the project where extern "C" fn is used. Other places that don't put it in `impl`...